sapphire/analysis/process_events.py

```diff
@@ -1190,3 +1195,53 @@ def __repr__(self):
                     (self.__class__.__name__, self.source_file.filename,
                      self.dest_file.filename, self.source_group._v_pathname,
                      self.dest_group._v_pathname, self.progress))
+
+
+class ProcessWeather(ProcessDataTable):
+
+    """Process HiSPARC weather to clean the data.
+
+    This class can be used to process a set of HiSPARC weather, to
+    remove duplicates and sort the data by timestamp to store it in to a
+    copy of the weather table.
+
+    """
+    table_name = 'weather'
+
+
+class ProcessWeatherFromSource(ProcessDataTableFromSource):
+
+    """Process HiSPARC weather from a different source.
+
+    This class is a subclass of ProcessWeather.  The difference is that in
+    this class, the source and destination are assumed to be different
+    files.  This also means that the source is untouched (no renaming of
+    original event tables) and the destination is assumed to be empty.
+
+    """
+    table_name = 'weather'
+
+
+class ProcessSingles(ProcessDataTable):
+
+    """Process HiSPARC singles data to clean the data.
+
+    This class can be used to process a set of HiSPARC singles data, to
+    remove duplicates and sort the data by timestamp to store it in to a
+    copy of the singles data table.
+
+    """
+    table_name = 'singles'
+
+
+class ProcessSinglesFromSource(ProcessDataTableFromSource):
+
+    """Process HiSPARC singles data from a different source.
+
+    This class is a subclass of ProcessSingles.  The difference is that in
```

Comment:
This is not true, same for the ProcessWeatherFromSource..
They are subclasses of `ProcessDataTableFromSource`.